geoh5py/data/visual_parameters.py

```diff
@@ -0,0 +1,146 @@
+#  Copyright (c) 2023 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import xml.etree.ElementTree as ET
+from warnings import warn
+
+from .text_data import TextData
+
+PARAMETERS = ["Colour"]
+ATTRIBUTES = ["tag", "text", "attrib"]
+
+
+class VisualParameters(TextData):
+    _xml: ET.Element | None = None
+
+    @property
+    def xml(self) -> ET.Element | None:
+        """
+        :obj:`str` XML string.
+        """
+        if self._xml is None and isinstance(self.values, str):
+            self.xml = self.values  # type: ignore
+
+        return self._xml
+
+    @xml.setter
+    def xml(self, values: str):
+        if not isinstance(values, str):
+            raise TypeError(f"Input 'values' for {self} must be of type {str}.")
+
+        try:
+            self._xml = ET.fromstring(values)
+        except ET.ParseError as exception:
+            raise ValueError(
+                f"Input 'values' for {self} must be a valid XML string ({exception})."
+            ) from exception
+
+    @property
+    def colour(self) -> None | str | dict:
+        """
+        :obj:`dict` the Colour dictionary.
+        """
+        return self.get_child("Colour", "text")
+
+    @colour.setter
+    def colour(self, values: str):
```

Comment:
I want to test the format of the colour to set (and same for other parameters we will add).
So no one could corrupt the xml file using those functions. 
Where could I get this information?

Reply:
@jincandescent Can you give us list or pattern for what's allowed by GA?